### PR TITLE
Add "post-test-infra-push-prowbazel" to push prowbazel image

### PR DESCRIPTION
**Pull request by clarketm**
/cc @PiotrSikora

### CODE REVIEW

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml [hunk 1]
```diff
@@ -62,6 +62,33 @@ postsubmits:
         - boskos
         - deploy
 
+  - name: post-test-infra-push-prowbazel
+    cluster: test-infra-trusted
+    run_if_changed: '^docker/prowbazel/(Dockerfile|entrypoint)$'
```
Comment (PiotrSikora) [inline prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: If you look at the last few prowbazel changes, you'll see that those files are not modified at all.

This should be triggered on changes to `docker/prowbazel/Makefile`, which bumps the version.

**Review by PiotrSikora — CHANGES_REQUESTED** [review]

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml (hunk 1 above)
Comment (clarketm) [reply prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: Okay, makes sense. Updated.

**Review by PiotrSikora — COMMENTED** [review]
Thanks! Looks good, but I don't know Prow itself enough to approve this.

**Review by PiotrSikora — COMMENTED** [review]
(clearing "request changes")

**Review by PiotrSikora — APPROVED** [review]

**Comment by clarketm** [comment]
@PiotrSikora - cool, thanks. I will assign someone from test-infra to review Prow changes.

**Review by cjwagner — APPROVED** [review]
/hold
LGTM assuming the push credentials have been added to the cluster.

**Comment by clarketm** [comment]
@cjwagner 
The push credentials have been created, added to cluster, and verified. 

/hold cancel

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml [hunk 2]
```diff
@@ -62,6 +62,33 @@ postsubmits:
...
+    branches:
+    - master
+    labels:
+      preset-prow-pusher-service-account: "true"
+      preset-dind-enabled: "true"
+    annotations:
+      testgrid-create-test-group: "false"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20191017-ac4b4b5-master
```
Comment (fejta) [inline prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: This seems like the wrong image. Why can't we use a standard istio image here?

**Review by fejta — COMMENTED** [review]
/cc @Katharine @cjwagner 

Is this the best pattern to use for this for now?

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml (hunk 2 above)
Comment (clarketm) [reply prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: This is the image I am familiar with for working with docker-in-docker. I will ask @howardjohn if there is an Istio image for this.

**Comment by howardjohn** [comment]
You can use the `build-tools` image that we use for almost everything. if
you run `entrypoint command` instead of `command` it will start up docker

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml (hunk 2 above)
Comment (clarketm) [reply prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: Updated to using the `gcr.io/istio-testing/build-tools` image.

**Pull request merged**